packages/core-entities/src/models/views/YieldViews.ts

```diff
@@ -430,6 +444,188 @@ export const YieldViews = (self: Instance<typeof NetworkModel>) => {
+      .reduce<string[]>((acc, symbol) => {
+        if (!acc.includes(symbol)) {
+          acc.push(symbol);
+        }
+        return acc;
+      }, []);
+
+    return uniqueUnderlyingSymbols;
+  };
+
+  const getDefaultHighestAPYSymbol = (productGroupData: any[][]) => {
+    const highestTotalAPYBySymbol: Record<string, number> = {};
+    productGroupData.flat().forEach((item) => {
+      const symbol = item?.underlying?.symbol;
+      if (symbol) {
+        const totalAPY = item.apy?.totalAPY || 0;
+
+        if (!highestTotalAPYBySymbol[symbol]) {
+          highestTotalAPYBySymbol[symbol] = totalAPY;
+        } else {
+          highestTotalAPYBySymbol[symbol] += totalAPY;
+        }
+      }
+    });
+
+      let highestAPYSymbol = '';
+      let highestAPYValue = 0;
+      for (const data in highestTotalAPYBySymbol) {
+        if (highestTotalAPYBySymbol[data] > highestAPYValue) {
+          highestAPYValue = highestTotalAPYBySymbol[data];
+          highestAPYSymbol = data;
+        }
+      }
+
+      return highestAPYSymbol;
+  };
+
+  const getPortfolioStateZeroBorrowData = () => {
+    const group = ['fCash', 'PrimeDebt'];
+    const productGroupData = group.map((group) => {
+      return getTokensByType(group)
+        .filter((data) => data?.isFCashDebt || data?.tokenType === 'PrimeDebt')
+        .map((t) => {
+          return {
+            token: t,
+            apy: getSpotAPY(t.id),
+            tvl: getTVL(t),
+            liquidity: getLiquidity(t),
+            underlying: t.underlying
+              ? getTokenByID(t.underlying)
+              : undefined,
+            collateralFactor: getDebtOrCollateralFactor(t, false),
+          };
+        });
+    });
+    const tokenList = getUniqueUnderlyingSymbols(productGroupData);
+    const defaultSymbol = 'ETH';
+    return {
+      tokenList,
+      productGroupData: productGroupData,
+      defaultSymbol,
+    };
+  };
+
+  const getPortfolioStateZeroEarnData = () => {
+    const group = ['fCash', 'PrimeCash', 'nToken'];
+    const productGroupData = group.map((group) => {
+      return getTokensByType(group)
+        .filter((data) => !data?.isFCashDebt)
+        .map((t) => {
+          return {
+            token: t,
+            apy: getSpotAPY(t.id),
+            tvl: getTVL(t),
+            liquidity: getLiquidity(t),
+            underlying: t.underlying
+              ? getTokenByID(t.underlying)
+              : undefined,
+            collateralFactor: getDebtOrCollateralFactor(t, false),
+          };
+        });
+    });
+    const tokenList = getUniqueUnderlyingSymbols(productGroupData);
+    const highestAPYSymbol = getDefaultHighestAPYSymbol(productGroupData);
+
+    return {
+      tokenList,
+      productGroupData,
+      defaultSymbol: highestAPYSymbol,
+    };
+  };
+
+  const getLeveragedNTokenData = () => {
+    const leveragedNTokenData = getTokensByType('nToken').map((t) => {
+          const debtTokens = getDefaultLeveragedNTokenAPYs(t);
```

Comment:
indentation here is wrong, you should get prettier installed in vscode and working. Otherwise if I touch a file it reformats the whole thing and it looks like there were a ton of changes.